openmetadata-service/src/main/java/org/openmetadata/service/security/auth/AuthenticatorHandler.java

```diff
@@ -0,0 +1,92 @@
+package org.openmetadata.service.security.auth;
+
+import static org.openmetadata.service.exception.CatalogExceptionMessage.NOT_IMPLEMENTED_METHOD;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import freemarker.template.TemplateException;
+import java.io.IOException;
+import java.util.UUID;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import org.openmetadata.schema.api.teams.CreateUser;
+import org.openmetadata.schema.auth.ChangePasswordRequest;
+import org.openmetadata.schema.auth.LoginRequest;
+import org.openmetadata.schema.auth.PasswordResetRequest;
+import org.openmetadata.schema.auth.RefreshToken;
+import org.openmetadata.schema.auth.RegistrationRequest;
+import org.openmetadata.schema.auth.TokenRefreshRequest;
+import org.openmetadata.schema.entity.teams.User;
+import org.openmetadata.schema.teams.authn.JWTAuthMechanism;
+import org.openmetadata.schema.teams.authn.JWTTokenExpiry;
+import org.openmetadata.service.auth.JwtResponse;
+import org.openmetadata.service.exception.CustomExceptionMessage;
+import org.openmetadata.service.security.jwt.JWTTokenGenerator;
+
+public interface AuthenticatorHandler {
+  JwtResponse loginUser(LoginRequest loginRequest) throws IOException, TemplateException;
+
+  void checkIfLoginBlocked(String userName);
+
+  void recordFailedLoginAttempt(User user) throws TemplateException, IOException;
+
+  void validatePassword(User storedUser, String reqPassword) throws TemplateException, IOException;
+
+  User lookUserInProvider(String userName);
+
+  default User registerUser(RegistrationRequest registrationRequest) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_FOUND, NOT_IMPLEMENTED_METHOD);
+  }
+
+  default void sendEmailVerification(UriInfo uriInfo, User user) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_FOUND, NOT_IMPLEMENTED_METHOD);
+  }
+
+  default void confirmEmailRegistration(UriInfo uriInfo, String emailToken) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_FOUND, NOT_IMPLEMENTED_METHOD);
```

Comment:
Its better to return 501 https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501 then the NOT_FOUND error code here